File: backend/src/modules/matricula/entity/Matricula.js
class Matricula {
  constructor({ id_matricula, data_inicio, data_fim, status, aluno_id_aluno, plano_id_plano }) {
    this.id_matricula = id_matricula;
    this.data_inicio = data_inicio;
    this.data_fim = data_fim;
    this.status = status;
    this.aluno_id_aluno = aluno_id_aluno;
    this.plano_id_plano = plano_id_plano;
  }
  validate() {
    const errors = [];
    if (!this.aluno_id_aluno) errors.push('Aluno é obrigatório');
    if (!this.plano_id_plano) errors.push('Plano é obrigatório');
    return errors;
  }
}
module.exports = Matricula;

File: backend/src/main/java/com/vitalativa/entity/Matricula.java
package com.vitalativa.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;

@Entity
@Table(name = "matricula")
public class Matricula {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_matricula")
    private Integer idMatricula;
    
    @Column(name = "data_inicio")
    private LocalDate dataInicio;
    
    @Column(name = "data_fim")
    private LocalDate dataFim;
    
    private String status;
    
    @Column(name = "aluno_id_aluno")
    private Integer alunoIdAluno;
    
    @Column(name = "plano_id_plano")
    private Integer planoIdPlano;
    
    public Matricula() {}
    
    public Integer getIdMatricula() { return idMatricula; }
    public void setIdMatricula(Integer idMatricula) { this.idMatricula = idMatricula; }
    public LocalDate getDataInicio() { return dataInicio; }
    public void setDataInicio(LocalDate dataInicio) { this.dataInicio = dataInicio; }
    public LocalDate getDataFim() { return dataFim; }
    public void setDataFim(LocalDate dataFim) { this.dataFim = dataFim; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public Integer getAlunoIdAluno() { return alunoIdAluno; }
    public void setAlunoIdAluno(Integer alunoIdAluno) { this.alunoIdAluno = alunoIdAluno; }
    public Integer getPlanoIdPlano() { return planoIdPlano; }
    public void setPlanoIdPlano(Integer planoIdPlano) { this.planoIdPlano = planoIdPlano; }
}